### PR TITLE
Add `net5` and `net6` as TargetFrameworks

### PR DESCRIPTION
**Pull request by oformaniuk**

### CODE REVIEW

File: source/Handlebars/Handlebars.csproj [hunk 1]
```diff
@@ -66,9 +66,24 @@
...
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)'=='net5'">
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)'=='net6'">
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+  </ItemGroup>
+
```
Comment (thompson-tomo) [inline source/Handlebars/Handlebars.csproj:76]: Microsoft.CSharp is a polyfill package hence this dependency should be able to be removed on the newer frameworks.

**Review by thompson-tomo — CHANGES_REQUESTED** [review]

**Comment by thompson-tomo** [comment]
Closes #573

**Comment by thompson-tomo** [comment]
@oformaniuk & @StefH I see that this has been sitting around for a while, any chance could the conflicts be resolved and then this merge progressed. :)

**Comment by oformaniuk** [comment]
> @oformaniuk & @StefH I see that this has been sitting around for a while, any chance could the conflicts be resolved and then this merge progressed. :)

I'd be happy to handle merging part if you can handle conflicts :)

File: source/Handlebars.Test/Handlebars.Test.csproj [hunk 2]
```diff
@@ -36,6 +36,14 @@
...
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(TargetFramework)'=='net5'">
+    <DefineConstants>$(DefineConstants);netcoreapp;netstandard;net</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)'=='net6'">
+    <DefineConstants>$(DefineConstants);netcoreapp;netstandard;net</DefineConstants>
+  </PropertyGroup>
+
```
Comment (thompson-tomo) [inline source/Handlebars.Test/Handlebars.Test.csproj:46]: Should be checked if this is really needed.

**Review by thompson-tomo — CHANGES_REQUESTED** [review]

**Comment by thompson-tomo** [comment]
@oformaniuk I will take a look later today to see if I can push changes to this branch.

**Comment by oformaniuk** [comment]
> @oformaniuk I will take a look later today to see if I can push changes to this branch. 

If not, it can be done by forking and merging back to this branch via PR 🙂

**Comment by thompson-tomo** [comment]
@oformaniuk pr #577 has been created to resolve the conflicts and implements the feedback i have above.

**Comment by StefH** [comment]
@thompson-tomo and @oformaniuk 
I think it's more clear that this PR is just closed and a new PR is created which only adds the new frameworks.

Side note: do we actually want to add .NET 5 because that one is out of support for some time? And instead just add .NET 6 and .NET 8 because .NET 7 is also almost out of official support?

**Comment by thompson-tomo** [comment]
@StefH i agree. What I will do tomorrow morning my time is cherry pick the commits on to a new branch from master & then submit that as a fresh pr & close my other pr.

**Comment by oformaniuk** [comment]
As agreed, change will be handled in a new PR.

**Comment by thompson-tomo** [comment]
New PR is #578 which is a nice and clean PR.

**Pull request closed without merging**